python/graphstorm/model/utils.py

```diff
@@ -311,6 +311,109 @@ def _exchange_node_id_mapping(local_rank, world_size, device,
         th.distributed.all_to_all(gather_list, data_tensors)
     return gather_list[0]
 
+def distribute_nid_map_to_embed(embeddings, local_rank, world_size, 
```

Comment:
the functions here should have unit tests.

Reply:
added in the new commit.